pdagent/__init__.py

```diff
@@ -30,5 +30,6 @@
 __version__ = "1.6"
 
 def enum(*sequential, **named):
-    enums = dict(zip(sequential, range(len(sequential))), **named)
+    # TODO(dobs): What are we trying to extract here?
+    enums = dict([x[::-1] for x in enumerate(sequential)], **named)
```

Comment:
This'll make quick & dirty enum -- you can call it like:

    enum(a, b, c, d)  # Enum({a: 0, b: 1, c: 2, d: 3})
    enum(a="asdf", b="bsdf", c="qwer")  # Enum({a: "asdf", b: "bsdf", c: "qwer"})
    enum(a, b, c=0, d=1) # Enum({a: 0, b: 1, c: 0, d: 1}) (it's not perfect)

`type` in this case builds [a simple class][1], something like a struct, out of dictionary arguments, so with the return value you'll be able to do stuff like:

  [1]: https://www.geeksforgeeks.org/python-type-function/

    Thing = enum(a, b, c, d)
    Thing.a # returns 0
    type(Thing) # returns <class 'type'>
    Thing.__name__ # returns "Enum"

instead of using dictionary semantics

Reply:
`x[::-1]` will reverse the list of sequential arguments -- not sure why that's necessary?

Reply:
Fixed the above a little -- realized after looking at usage that the positional arguments should be the names that you want 😅 

Could really use anything that can be used as a dictionary key though (anything that implements the hashable functions I think)

Reply:
Yeah, definitely room for improvement here — would like to rip out a lot of the custom enum stuff possibly in favour of using Enum directly.

Reply:
:nod: Enum is new to Python 3 iirc, makes sense they weren't using it before